In [1]:
import numpy as np 
import pandas as pd 

READING THE DATASET

In [4]:
data=pd.read_csv("sample_sales_data.csv")
data.head()
data.drop(columns=["Unnamed: 0"], inplace=True, errors='ignore')
data.head()

transaction_id            timestamp  \
0  a1c82654-c52c-45b3-8ce8-4c2a1efe63ed  2022-03-02 09:51:38   
1  931ad550-09e8-4da6-beaa-8c9d17be9c60  2022-03-06 10:33:59   
2  ae133534-6f61-4cd6-b6b8-d1c1d8d90aea  2022-03-04 17:20:21   
3  157cebd9-aaf0-475d-8a11-7c8e0f5b76e4  2022-03-02 17:23:58   
4  a81a6cd3-5e0c-44a2-826c-aea43e46c514  2022-03-05 14:32:43   

                             product_id category customer_type  unit_price  \
0  3bc6c1ea-0198-46de-9ffd-514ae3338713    fruit          gold        3.99   
1  ad81b46c-bf38-41cf-9b54-5fe7f5eba93e    fruit      standard        3.99   
2  7c55cbd4-f306-4c04-a030-628cbe7867c1    fruit       premium        0.19   
3  80da8348-1707-403f-8be7-9e6deeccc883    fruit          gold        0.19   
4  7f5e86e6-f06f-45f6-bf44-27b095c9ad1d    fruit         basic        4.49   

   quantity  total payment_type  
0         2   7.98     e-wallet  
1         1   3.99     e-wallet  
2         2   0.38     e-wallet  
3         4   0.76     e-wallet  
4         2   8.98   debit card

DESCRIPTIVE ANALYSIS

In [5]:
data.describe()

unit_price     quantity        total
count  7829.000000  7829.000000  7829.000000
mean      7.819480     2.501597    19.709905
std       5.388088     1.122722    17.446680
min       0.190000     1.000000     0.190000
25%       3.990000     1.000000     6.570000
50%       7.190000     3.000000    14.970000
75%      11.190000     4.000000    28.470000
max      23.990000     4.000000    95.960000

- **Count:**
  - `unit_price`: 7829 records
  - `quantity`: 7829 records
  - `total`: 7829 records

- **Mean (Average):**
  - `unit_price`: The average unit price is approximately $7.82.
  - `quantity`: The average quantity is approximately 2.50.
  - `total`: The average total is approximately $19.71.

- **Standard Deviation (Std):**
  - `unit_price`: The standard deviation of unit prices is approximately $5.39.
  - `quantity`: The standard deviation of quantities is approximately 1.12.
  - `total`: The standard deviation of totals is approximately $17.45.

- **Minimum (Min):**
  - `unit_price`: The minimum unit price is $0.19.
  - `quantity`: The minimum quantity is 1.
  - `total`: The minimum total is $0.19.

- **25th Percentile (Q1):**
  - `unit_price`: 25% of the unit prices are below $3.99.
  - `quantity`: 25% of the quantities are 1.
  - `total`: 25% of the totals are below $6.57.

- **50th Percentile (Median):**
  - `unit_price`: 50% of the unit prices are below $7.19.
  - `quantity`: 50% of the quantities are below 3.
  - `total`: 50% of the totals are below $14.97.

- **75th Percentile (Q3):**
  - `unit_price`: 75% of the unit prices are below $11.19.
  - `quantity`: 75% of the quantities are below 4.
  - `total`: 75% of the totals are below $28.47.

- **Maximum (Max):**
  - `unit_price`: The maximum unit price is $23.99.
  - `quantity`: The maximum quantity is 4.
  - `total`: The maximum total is $95.96.

These statistics provide a summary of the central tendency, dispersion, and distribution of the three variables in your dataset.

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7829 entries, 0 to 7828
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   transaction_id  7829 non-null   object 
 1   timestamp       7829 non-null   object 
 2   product_id      7829 non-null   object 
 3   category        7829 non-null   object 
 4   customer_type   7829 non-null   object 
 5   unit_price      7829 non-null   float64
 6   quantity        7829 non-null   int64  
 7   total           7829 non-null   float64
 8   payment_type    7829 non-null   object 
dtypes: float64(2), int64(1), object(6)
memory usage: 550.6+ KB


output is a summary of the structure and data types of a DataFrame using pandas in Python. Here's an interpretation of the information provided:

- **RangeIndex:**
  - The DataFrame has 7829 entries (rows).
  - The RangeIndex is from 0 to 7828.

- **Data columns (total 9 columns):**
  1. `transaction_id`: Object data type with 7829 non-null entries.
  2. `timestamp`: Object data type with 7829 non-null entries.
  3. `product_id`: Object data type with 7829 non-null entries.
  4. `category`: Object data type with 7829 non-null entries.
  5. `customer_type`: Object data type with 7829 non-null entries.
  6. `unit_price`: Float64 data type with 7829 non-null entries.
  7. `quantity`: Int64 data type with 7829 non-null entries.
  8. `total`: Float64 data type with 7829 non-null entries.
  9. `payment_type`: Object data type with 7829 non-null entries.

- **Dtypes:**
  - Float64: Columns `unit_price` and `total` are of type float.
  - Int64: Column `quantity` is of type integer.
  - Object: Columns `transaction_id`, `timestamp`, `product_id`, `category`, `customer_type`, and `payment_type` are of type object (usually string).

- **Memory usage:**
  - The DataFrame occupies approximately 550.6 KB in memory.

This information is useful for understanding the structure of your dataset, including the number of entries, data types, and memory usage for each column. It also indicates whether there are any missing values (non-null counts) in each column.

In [7]:
data.isnull().sum()

transaction_id    0
timestamp         0
product_id        0
category          0
customer_type     0
unit_price        0
quantity          0
total             0
payment_type      0
dtype: int64

ABOVE INDICATES THAT NO NULL VALUES IN THE DATASET